### ISSUE TITLE
Enable lighting-related quests during daylight; at least optionally

### ISSUE THREAD

**Issue by pigsonthewing** [streetcomplete/StreetComplete#3248]
Please revert the change which made several lighting-related quests only show during hours of darkness; or at least make it possible for a user to override that change in their settings.

It is often possible to tell that, for example, a path through a park or countryside is not it, by the complete absence of lighting posts or other infrastructure. Likewise simple bus stops with no lamp.

Similarly, in built up areas the presence of maintained lighting posts, even in daylight, shows that the way is lit.

**Comment by westnordost** [comment]
It will not be reverted.

For many ways, especially those ways that are near or alongside streets that are lit, it is not possible to say with certainty whether the street lighting reaches to these well. Because quests in StreetComplete should be easily answerable, they are now only shown after sundown.

> Likewise simple bus stops with no lamp.

There may be a street lantern right at it or near it. To judge whether the bus stop is thus lit can only be ascertained with certainty when the lights are actually on.

**Comment by pigsonthewing** [comment]
> It will not be reverted.

Then, as requested, please make it user-configurable.

 > For many ways, especially those ways that are near or alongside streets that are lit, it is not possible to say with certainty whether the street lighting reaches to these well.

And for many ways it is possible to see that there **is no street lighting** anywhere in the vicinity.

> > Likewise simple bus stops with no lamp.
> 
> There may be a street lantern right at it or near it. To judge whether the bus stop is thus lit can only be ascertained with certainty when the lights are actually on.

I referred to cases where there is **no lamp**. The lights cannot be on if they do not exist.

**Comment by mnalis** [comment]
I acknowledge that both variants (always showing a `lit`, and only showing it during the night) have their own disadvantages. And I understand that for average StreetComplete user the new behaviour is likely better in majority of cases.

But I still miss the ability to mark forest paths (with no infrastructure anywhere) as unlit. Anyway, I'd also prefer if it was "only show during night" by default, but the users had the option to override it in the menu. But I also do understand the need (for StreetComplete especially!) to keep UI as simple as possible.

For reference, see original discussion in https://github.com/streetcomplete/StreetComplete/issues/1285

**Comment by mnalis** [comment]
I must say I'm afraid @davidpnewton does have a point, this does seem to promote non-onsite mapping. 

Is there some solution that might work?

- add a preference to choose if such limiting of quest  (perhaps like per-quest settings in @Helium314 fork? maybe only show if "expert mode" checkbox is on - https://github.com/streetcomplete/StreetComplete/discussions/3003 so no extra complexity would be shown for "simple" default users) 
- modify `lit` quest so it asks for `lit` status always if the `highway=path` or `highway=track` with `surface != asphalt` (eg. probably not in the city). 
  Alternatively use day/night filter for it only if `highway =~ motorway|trunk|primary|secondary|tertiary|residental`
- have two `lit` quests, one 24/7, and one only on night, so users can enable what they want in their preset (with night-only quest being the one selected by default)
- or at least do not add `source=survey` to changeset tags if user is more than 300m from GPS location (or GPS location is unknown). That wouldn't discourage offsite mapping, but would at least (in a reasonable way) drop the pretense that SC is being used only for on-site mapping. (also related: https://github.com/streetcomplete/StreetComplete/issues/3137#issuecomment-894872930)

**Comment by matkoniecz** [comment]
> So you want people to hang round in dark forests after dark do you?

If you live in unsafe area and that is dangerous in your area or for you, please do not do this. As reminder, on the first opening app asks to not do dangerous stuff just to map.

> So that's in line with your possible want two: encouraging people to tag things they are not physically near at the time of tagging.

No. Note that if you start doing this app will ask you to do doing this.

> Aren't the results of silly removals of functionality great!

This quest was quite confusing in day time, in multiple cases people were entering wrong data

> then one of those two general points must be true

This misses several possibilities. Also, if I would not be on mobile I would insert image of xkcd with spacebar heating.

**Comment by mnalis** [comment]
This one https://xkcd.com/1172/ ? xkcd reference is always nice, but I do not think that users wanting to map _"no lights in forests path via SC"_ is really such a _horrible_ idea. (actually, user having to change the system clock to workaround it seems like much worse workflow idea to me.)

**Comment by Helium314** [comment]
> So you want people to hang round in dark forests after dark do you?

I suspect whatever dark forest you mean might be tagged quite unusual if StreetComplete asks whether the ways there are lit.

**Comment by mnalis** [comment]
> I suspect whatever dark forest you mean might be tagged quite unusual if StreetComplete asks whether the ways there are lit.

@Helium314 what do you mean? I can easily see `highway=track`+`surface=compacted`  (or `highway=path`) going through `natural=wood` (or `landuse=forest`) area and being marked with `lit=yes` or `lit=no`. What is so _unusual_ about it?

**Comment by Strubbl** [comment]
I really do not like that the lit question is only shown during night. And i cannot change this, _in my opinion_ stupid, default behavior of the app? I am thinking how to work around this option during day or i am not going to answer this question at all anymore.

For many places answering questions only during night or day would mean visiting the same place multiple times. Often i do not have the chance to do so. And often i do not want to do that during night. There are reasons.

> There may be a street lantern right at it or near it. To judge whether the bus stop is thus lit can only be ascertained with certainty when the lights are actually on.

How shall i measure the luminous intensity? And what is the treshold value to put a lit=yes or lit=no?

**Comment by westnordost** [comment]
Hm so you think that limiting the quest to only after sundown was wrong altogether and should be reverted?

We can talk about that, but please first take the time to read through #1285 (and linked) where it was discussed.

**Comment by Strubbl** [comment]
> Hm so you think that limiting the quest to only after sundown was wrong altogether and should be reverted?

From my point of view, yes, we could save these lines of code. But that is only my preference. There is always somebody who has arguments for or against that behavior of the app. It would be nice, if the user can decide here. And of course the current state (v34.1) can be the default state. 

So i read the other thread. I am answering here.

The argument with creating a profile for day/night was already mentioned and i second this. This solution is far better than some automagic. @mnalis comment in https://github.com/streetcomplete/StreetComplete/issues/1285#issuecomment-897004519 has a lot of good arguments.

There was an argument:

> In some cases check during the night is necessary to check whatever cycleway/footway next to the road is also lit

This is a special case. If i cannot reliably answer the lit quest, i can decide by myself to not answer it and let somebody (maybe myself) check it during night. Yes, sometimes this means i have to skip a lot of lit quests. 

> My intention was to make collecting data more efficient, not to prevent an incorrect data.

It is not efficient in the case, that i can reliably answer this quest during day where i am sure ways are lit/not lit and keep the quest open for ways i am not sure. Those i can efficiently survey during night, because the obvious ones are answered already.

**Comment by peternewman** [comment]
I've just realised, don't we need to disable the speed quest during the daytime too now (or at least for residential roads)? In the UK at least it asks (and tags) if the road is lit in some circumstances as that affects the speed limit. If we're saying we can't be trusted to tag it directly in the daytime, that should be the case for indirect tagging too.

FWIW like @mnalis I've had many paths through woods and fields where I'm asked if they're lit, and sometimes one major one has lamp posts and other minor ones don't, so I can't now answer those quests sensibly, and I'm unlikely to be in most of those areas in the dark. I'm also encountering areas that are lit during the daytime (i.e. 24/7) which I can't tag as such, I still don't understand how someone at night will know that status (or be expected to remember both), so they're probably getting miss-tagged (although part of that is down to a tagging scheme that includes two states you can't observe at the same time in the same tag, rather than say lit:daytime=yes, lit:nighttime=yes).

**Comment by arrival-spring** [comment]
> I've just realised, don't we need to disable the speed quest during the daytime too now (or at least for residential roads)? In the UK at least it asks (and tags) if the road is lit in some circumstances as that affects the speed limit. If we're saying we can't be trusted to tag it directly in the daytime, that should be the case for indirect tagging too.

I believe that the speed limit is affected by the presence (or not) of street lighting, and not whether or not the lighting is in use (although it almost always will be).

Perhaps the followup question could be reworded and it only adds `lit=no`and not `lit=yes` even if there are street lights.

**Comment by westnordost** [comment]
I am thinking about completely reverting this feature. Via the feedback function from within the app, I now got several requests to remove that feature again. I can also follow the arguments that whether anywhere are lanterns or not is visible during day as well.

After all, everyone can now with quest profiles create a day and a night-profile.

**Comment by matkoniecz** [comment]
Main problem is that in showing SC to other people there were multiple cases where people made opposite mistake - they failed to notice existing lanterns.

Variants included:

- steps lit by high lantern quite far away
- light that was positioned at low level (placed about 40cm or 1m from ground)
- footway lit by street lights quite far away

And general confusion "how I am supposed to know which version is correct"?

Would it be maybe solvable by adding one more quest (disabled by default) that would ask about `lit` status also during day?
-

**Comment by mnalis** [comment]
Having two quests `lit (night only)` and `lit (always ask)` with just the first one enabled by default would work for me, @matkoniecz. From cleanliness point of view, it does look like the kludge, however. 

Perhaps per-quest configure button (like in @Helium314 fork) would look cleaner (and could also be used for other problematic quests in the future)? (those two were my favourites, although I did suggest few other workarounds around most glaring issues at https://github.com/streetcomplete/StreetComplete/issues/3248#issuecomment-914795588)

Failing that, @westnordost idea of falling back to previous behaviour does look like lesser of two evils.

**Comment by smichel17** [comment]
> From cleanliness point of view, it does look like the kludge, however

From my point of view, it looks a little like [subquests](https://github.com/streetcomplete/StreetComplete/discussions/3209#discussioncomment-1316792)

**Comment by matkoniecz** [comment]
> Perhaps per-quest configure button

That is going to be way more complex to implement and maintain

**Comment by mnalis** [comment]
> > Perhaps per-quest configure button
> That is going to be way more complex to implement and maintain (at least I think so)

Perhaps, but (to my untrained eye) current proof-of-concept implementation does not look that bad - see for example https://github.com/Helium314/StreetComplete/commit/e256a94617d6465db3a6cef92b89ef83e6665d35 or https://github.com/Helium314/StreetComplete/commit/a60a389024f7db64091b25c950e511180c3a5ff9. 

(Of course only the few quests that do require possible configuration get that configure icon, as can be seen in that [fork](https://github.com/Helium314/StreetComplete/releases))

**Comment by Helium314** [comment]
a) I know someone who _really_ dislikes adding more settings and made that very clear
b) The way these settings are implemented in my fork mixes android-specific stuff into the quest code. This will not be accepted in SC, and fixing/adjusting this will likely be more work.
c) My fork has the option of disabling the daytime filter (affects all quests that use it) instead of using the per-quest settings, https://github.com/Helium314/StreetComplete/commit/965a9427c4e0cf344cfd0e809fb9f43639da1bee

**Comment by westnordost** [comment]
It is an interesting idea. I thought about this but reached to the conclusion that this is not a good solution.

The reason why the lit quest has issues during daytime is because for some/many ways, it is really difficult to answer if a particular segment of way can be considered lit. So, during daytime, sometimes the "🐿️ Easy answer" rule does not apply.

Adding a confirmation dialog does not solve this problem, instead it just makes contributing this information during daytime more pesky for *all* lit quests, even those for which a clear and easy answer is possible.

**Comment by westnordost** [comment]
Alright, so I reverted the disablement of certain quests during daytime. When it was implemented, we had
- no quest presets. So, no possibility to e.g. create a "night jog" preset. This changed.
- quests were ordered by "importance" rather than how easy and quick they can be solved. So, I think it was a problem that a lit quest one wasn't able to answer blocked other quests. Now, I moved the lit quests down in the default quest order so everything else is asked first.

I played with the idea to default-disable the quest with an explanation message like "Sometimes it is hard to ascertain during daytime whether a particular section of path is lit. Consider to only enable this quest when the lights are on."
However, I don't remember the lit quests to be **that** problematic. It is an option for the future though, in case the lit quest turns out to be still somewhat problematic in this way.

**Comment by westnordost** [comment]
Alternative would have been, by the way, that "show quest only during nighttime" setting would only have the effect that during day, the quest is considered as "disabled by default". So the user can in the settings enable this quest type manually. So during night, the quest would be enabled and during day, the user can manually enable it after acknowledging an explanation message.

But this is somewhat of a incomprehensible magic behavior. Then rather unconditionally disable the quest by default, see my last post.

**Comment by RubenKelevra** [comment]
Well, I can see both sides. I guess we just need a more specific quest which does use an additional tag. 

Sometimes said street lights are extremely bright and sometimes areas are barely lit between two street lamps.

This cannot be surveyed at day time.

If we add an additional quest for the quality of the illumination, a user might spot the very dark spots and answer them with either very poorly lit or unlit.

There's the tag "lit:perceived" which offers exactly this.

**Comment by mnalis** [comment]
@RubenKelevra yes, [lit:perceived](https://wiki.openstreetmap.org/wiki/Key:lit:perceived) quest, if implemented, would be a great candidate for (likely disabled-by-default as not to overload regular SC users?) night-only quest (and it could be used to override answer from `lit` all-day quest if needed, thus alleviating outlined potential problems with precision of "daytime mappers of lit quest"). 

Judging by it's [wiki](https://wiki.openstreetmap.org/wiki/Key:lit:perceived) it looks only mildly subjective,  and has about 1800 uses so far, so perhaps it might be an option.

**Comment by RubenKelevra** [comment]
> @RubenKelevra yes, [lit:perceived](https://wiki.openstreetmap.org/wiki/Key:lit:perceived) quest, if implemented, would be a great candidate for (likely disabled-by-default as not to overload regular SC users?) night-only quest (and it could be used to override answer from `lit` all-day quest if needed, thus alleviating outlined potential problems with precision of "daytime mappers of lit quest"). 

I don't think we overload regular users. Never heard that complaint from anyone... its more a disappointment when someone cannot solve something, like 'oh I can't even possibly see the roof of that building clearly' and then proceeds to wander around with a long neck awkwardly around the house 😂

So I usually disable those building quests for new users. Especially the house type quest is kind of confusing.

So as long as the quest is solvable 99% of the time by easy observation without any explanation from me, it's well designed IMHO. This would be the case for a perceived lit one - nice graphics presupposed. But @westnordost never failed to deliver on that :) 


> Judging by it's [wiki](https://wiki.openstreetmap.org/wiki/Key:lit:perceived) it looks only mildly subjective,  and has about 1800 uses so far, so perhaps it might be an option.

Well, I think it's pretty good actually. Except that colorless tag.

It is oddly placed in a tag which only covers the amount of light. A fixture might spend very poor amounts of yellow light, or a huge amount. This tag is completely misplaced IMHO since you cannot tag both.

It assumes that yellow light is always perfect, which is not the case.

I think this should maybe moved to some other subtag, like "lit:type" or something like that.

This tag was proposed but never made it through quality control:

https://wiki.openstreetmap.org/wiki/Proposed_features/Key:lit:type

---

tldr: just create a perception tag quest (but exclude the colorless value).

**Comment by smichel17** [comment]
> Especially the house type quest is kind of confusing.

Perhaps you could create a new discussion around how to improve it? Building type is currently necessary in order to unlock the address quest.

**Comment by mnalis** [comment]
- @RubenKelevra As for the `lit:perceived`, due to some amount of overlap, it might even be preferred to merge it into `lit` quest, in a way that during the day quest is asked as before (just basic answers `yes`/`no`/...), but during the night extra answers appear (for filling in `lit:perceived` data) so more detailed answers can be provided. (Perhaps extra detailed answers should also be shown during the day, but grayed-out/disabled, for giving users indication they could be answering more here).  

  I'm not too keen about proposed `lit:type` (nor even more ultra-detailed https://wiki.openstreetmap.org/wiki/Proposed_features/Key:light_source with its sets of tags). `lit:perceived` would add just enough information IMHO, even if it does not distinguish between all possible lighting types - that non-detailedness is IMHO what makes it _**easily solvable**_. Especially since all artificial lightning is sub-ideal (ie. having much worse [CRI](https://en.wikipedia.org/wiki/Color_rendering_index) than during daylight), making a fuss about cold-white LED vs. halogen vs. xenon is hopeless. I'd even separate `llt:perceived=colorless` into `Other answers` instead of being offered in main choice picker.
 
- As for the mentioned `house type` quest, also note that it was suggested that `House Address` quest should be solvable even without solving type of the building first, if it's priority was changed (or `building type` quest disabled). It was being considered, but is currently waiting for more input/experience and/or implementation of Layers  - see https://github.com/streetcomplete/StreetComplete/issues/2464#issuecomment-782962059 and below.

**Comment by RubenKelevra** [comment]
> - @RubenKelevra As for the `lit:perceived`, due to some amount of overlap, it might even be preferred to merge it into `lit` quest, in a way that during the day quest is asked as before (just basic answers `yes`/`no`/...), but during the night extra answers appear (for filling in `lit:perceived` data) so more detailed answers can be provided. (Perhaps extra detailed answers should also be shown during the day, but grayed-out/disabled, for giving users indication they could be answering more here).  

Well, I wouldn't keep them separate. 

- "Are there street lamps?"

- "How lit is this way at night?"

And if the answers are for example 'yes' and 'none/minimal' we could show a popup asking if the lights are broken or not existing for example. 

>   I'm not too keen about proposed `lit:type` (nor even more ultra-detailed https://wiki.openstreetmap.org/wiki/Proposed_features/Key:light_source with its sets of tags). `lit:perceived` would add just enough information IMHO, even if it does not distinguish between all possible lighting types - that non-detailedness is IMHO what makes it _**easily solvable**_. Especially since all artificial lightning is sub-ideal (ie. having much worse [CRI](https://en.wikipedia.org/wiki/Color_rendering_index) than during daylight), making a fuss about cold-white LED vs. halogen vs. xenon is hopeless. I'd even separate `llt:perceived=colorless` into `Other answers` instead of being offered in main choice picker.

Colorless just doesn't fit well and is barely even used. I would avoid implementing it and move on. 🤷‍♂️

Maybe even start a discussion on the wiki page to see if others think the same (for example the author(s)) or users who used the tag before. 



> - As for the mentioned `house type` quest, also note that it was suggested that `House Address` quest should be solvable even without solving type of the building first, if it's priority was changed (or `building type` quest disabled). It was being considered, but is currently waiting for more input/experience and/or implementation of Layers  - see https://github.com/streetcomplete/StreetComplete/issues/2464#issuecomment-782962059 and below.


Well it feels just too tedious to decide between detached house, house, apartment etc.

I feel we need just need to put in a value instead, which flags the building as inhabited: residential. Everything else is quite difficult to observe sometimes.

Additionally it would be nice to have the information about additional uses of the building, like retail or a amenity etc in the floor level.

So maybe make it multiple choice, like "people are living there" and it has a shop in it and make `building=residential; retail` out of it or something like that.

This would open the possibility to create quests for missing shop nodes for example, since we now have the information that there should be a shop in a building.

**Comment by smichel17** [comment]
@RubenKelevra I like the direction of your ideas, but can you please post them in a **new [discussion](https://github.com/streetcomplete/StreetComplete/discussions/new) or [issue](https://github.com/streetcomplete/StreetComplete/issues/new?assignees=&labels=&template=feature_request.md)**?

This conversation is going to get lost here, in a closed issue about the `lit` quest.

**Comment by matkoniecz** [comment]
> Well, I would keep them separate.
>
>    "Are there street lamps?"
>
>    "How (well) lit is this way at night?"

Note that we would need an established or accepted tagging for that.

> This would open the possibility to create quests for missing shop nodes for example, since we now have the information that there should be a shop in a building.

`building=retail` may not house shop anymore (repurposed, abandoned, under reconstruction)...

See #3398

**Comment by RubenKelevra** [comment]
> > Well, I would keep them separate.
> > "Are there street lamps?"
> > "How (well) lit is this way at night?"
> 
> Note that we would need an established or accepted tagging for that.

We have. Street lamps along the street are `lit=yes`, while how well the way is lit at night is `lit:perceived=*`.




> > This would open the possibility to create quests for missing shop nodes for example, since we now have the information that there should be a shop in a building.
> 
> `building=retail` may not house shop anymore (repurposed, abandoned, under reconstruction)...

True, but that would still be worth a shop node IMHO, with either `shop=vacant` or `disused:shop=yes`.

**Comment by westnordost** [comment]
1800 usages worldwide, no proposal, status unspecified is hardly established tagging, not even with a lot of goodwill.
Plus, if this was a proposal, I doubt it would be accepted.

**Comment by RubenKelevra** [comment]
@westnordost is there an alternative?

**Comment by westnordost** [comment]
I don't think so.

**Comment by Bubu** [comment]
I just wanted to share a slightly different opinion here, which I don't think has been mentioned before:

I quite liked these quests being only being enabled at night, it gave you something new to do at different times of the day and in that made StreetComplete being played as a game, instead of being used as a tool to fill OSM, more engaging. (Which means I got quite a few people to play that have had no interaction with openstreetmap mapping whatsoever before. "Let's go back to this area at night, to discover new quests" really works quite well for the discovery and adventure aspect of it :))

Just wanted to throw this in here. This doesn't mean that reverting this isn't the right call to make for now.

**Comment by TurnrDev** [comment]
Hi, long time no speak. I wrote the day/night stuff, and whilst it's sad to see it gone, I agree that it's just not needed now we have quest presets. I do think there should be better tagging of lit:perceived=* but someone needs to write up a proposal for that first. Dibs not me. Keep up the good work @westnordost and @matkoniecz
